[FEAT] Add doc to simplify podspecs for Native Modules and Native Components (#3321)
facebook/react-native-website: docs/the-new-architecture/backward-compatibility-fabric-components.md

The Apple platform installs Fabric Native Components using [Cocoapods](https://cocoapods.org) as a dependency manager.

> If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/main/scripts/react_native_pods.rb#L145) function, **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoid them when it is not enabled.

Author: In theory, we could. This change will land in 0.71, so everyone will be in the "Otherwise" scenario until we release that version. Also, the upgrade tool won't be of any help in this case because we are targeting libraries and there is no such tool for libraries (the upgrade helper only works on the App Template).

That's why I think it is valuable to leave the manual instructions, at least for version 0.71.0.
What do you think?

> If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/main/scripts/react_native_pods.rb#L145) function, **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoid them when it is not enabled.

Reviewer: Hm, I'm still confused - who won't be using `install_module_dependencies`?

> If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/main/scripts/react_native_pods.rb#L145) function, **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoid them when it is not enabled.

Author: > who won't be using install_module_dependencies

Currently, some libraries have already migrated, so it is probably not worth migrating again to `install_module_dependencies`. 
The other case that come to my mind are libraries which uses folly directly and that may need to hard-pin to a specific folly version (e.g.: reanimated, as far as I know).

But I think that it is an edge case.

> If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/82e9c6ad611f1fb816de056ff031716f8cb24b4e/scripts/react_native_pods.rb#L145) function, **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoid them when it is not enabled.

Reviewer 2: ```suggestion
If you are already using the [`install_module_dependencies`](https://github.com/facebook/react-native/blob/82e9c6ad611f1fb816de056ff031716f8cb24b4e/scripts/react_native_pods.rb#L145) function, then **there is nothing to do**. The function already takes care of installing the proper dependencies when the New Architecture is enabled and avoiding them when it is not enabled.
```

end
```

> You want to install the extra dependencies when the New Architecture is enabled and you want to avoid installing them when it's not.

Reviewer 2: ```suggestion
You should install the extra dependencies when the New Architecture is enabled, and avoid installing them when it's not.
```
